components/director/internal/healthz/handler.go

```diff
@@ -1,17 +1,33 @@
 package healthz
 
 import (
+	"context"
+	"fmt"
 	"net/http"
 
 	"github.com/pkg/errors"
 	"github.com/sirupsen/logrus"
 )
 
+//go:generate mockery -name=Pinger -output=automock -outpkg=automock -case=underscore
+type Pinger interface {
+	PingContext(ctx context.Context) error
+}
+
 // NewHTTPHandler returns function which handles healtz calls
-func NewHTTPHandler(log *logrus.Logger) func(writer http.ResponseWriter, request *http.Request) {
+func NewHTTPHandler(p Pinger, log *logrus.Logger) func(writer http.ResponseWriter, request *http.Request) {
 	return func(writer http.ResponseWriter, request *http.Request) {
+		err := p.PingContext(request.Context())
+		if err != nil {
+			writer.WriteHeader(http.StatusInternalServerError)
+			_, err = writer.Write([]byte(fmt.Sprintf("error: %v", err)))
```

Comment:
I'm wondering if we should log this error as well. Technically we can get k8s events and gather issues from there, but... We can have all in one place (Director logs) as well. What do you think?